keyboards/skyloong/gk68/pro/pro.c

```diff
@@ -0,0 +1,242 @@
+            if (!rgb_matrix_get_flags()) {
+               RGB_MATRIX_INDICATOR_SET_COLOR(MAC_MOD_INDEX, 0, 0, 0);
+            }
+            }else{
+                if (!rgb_matrix_get_flags()) {
+                   RGB_MATRIX_INDICATOR_SET_COLOR(WIN_MOD_INDEX, 0, 0, 0);
+                 }
+              }
+         } break;
+
+     case 1:{
+       if (L_MAC) {
+            RGB_MATRIX_INDICATOR_SET_COLOR(MAC_MOD_INDEX, 255, 255, 255);
+            if (!rgb_matrix_get_flags()) {
+               RGB_MATRIX_INDICATOR_SET_COLOR(WIN_MOD_INDEX, 0, 0, 0);
+            }
+            }else{
+                if (!rgb_matrix_get_flags()) {
+                   RGB_MATRIX_INDICATOR_SET_COLOR(MAC_MOD_INDEX, 0, 0, 0);
+                 }
+              }
+         } break;
+
+      default:{
+         if (!rgb_matrix_get_flags()) {
+            RGB_MATRIX_INDICATOR_SET_COLOR(WIN_MOD_INDEX, 0, 0, 0);
+            RGB_MATRIX_INDICATOR_SET_COLOR(MAC_MOD_INDEX, 0, 0, 0);
+         }
+      }
+    }
+    return false;
+}
+
+
+void suspend_power_down_kb(void) {
+    writePinLow(SDB);
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb(void) {
+    writePinHigh(SDB);
+    suspend_wakeup_init_user();
+}
+bool shutdown_kb(bool jump_to_bootloader) {
+    writePinLow(SDB);
+    return shutdown_user(jump_to_bootloader);
+}
+#endif
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    if (!process_record_user(keycode, record)) {
+        return false;
+    }
+    switch (keycode) {
+#    ifdef RGB_MATRIX_ENABLE
+        case RGB_TOG:
+            if (record->event.pressed) {
+                switch (rgb_matrix_get_flags()) {
+                    case LED_FLAG_ALL: {
+                        rgb_matrix_set_flags(LED_FLAG_NONE);
+                        rgb_matrix_set_color_all(0, 0, 0);
+                    } break;
+                    default: {
+                        rgb_matrix_set_flags(LED_FLAG_ALL);
+                    } break;
+                }
+            }
+         return false;
+#    endif
+     case TO(0):
+      if (record->event.pressed) {
+       L_WIN = 1;
+       set_single_persistent_default_layer(0); // Save default layer 0 to eeprom
+      } else {
+       L_WIN = 0;
+      }
+      return true; // continue all further processing of this key
+
+     case MO(2):
+      if (record->event.pressed) {
+       FN_WIN = 1;
+      } else {
+       FN_WIN = 0;
+      }
+      return true; // continue all further processing of this key
+
+     case TO(1):
+      if (record->event.pressed) {
+       L_MAC = 1;
+       set_single_persistent_default_layer(1);  //Save default layer 1 to eeprom
+      } else {
+       L_MAC = 0;
+      }
+      return true; // continue all further processing of this key
+
+     case MO(3):
+      if (record->event.pressed) {
+       FN_MAC = 1;
+      } else {
+       FN_MAC = 0;
+      }
+      return true; // continue all further processing of this key
+
```

Comment:
Personally I do not think this code should exist at the keyboard level. While it can be overridden, it makes too many assumptions on the running keymap and some questionable behaviour changes to layer keys.

Reply:
These are mainly to improve the user experience. It can allow the user to have relevant instructions when switching layers, and can restore the normal light effect state after the completion of the operation.

Reply:
The issue isnt that i have failed to understand what the code does.

Reply:
I would recommend placing this code in a vendor keymap instead (`default_<vendorname>`), leaving the keyboard level and `default` keymap pristine.

Reply:
Ok, thanks for your advice. I'll try to change it.

Reply:
Is it  correct? Would you please check it again? @fauxpark  @zvecr